benchmarks/graphics_pipeline/main.cpp

```diff
@@ -240,22 +255,62 @@ class ProjApp
 
 void FreeCamera::Move(MovementDirection dir, float distance)
 {
+    // Given that v = (1, mTheta, mPhi) is where the camera is looking at in the Spherical
+    // coordinates and moving forward goes in this direction, we have to update the
+    // camera location for each movement as follows:
+    //      FORWARD:     distance * unitVectorOf(v)
+    //      BACKWARD:    -distance * unitVectorOf(v)
+    //      RIGHT:       distance * unitVectorOf(1, mTheta + pi/2, pi/2)
+    //      LEFT:        -distance * unitVectorOf(1, mTheta + pi/2, pi/2)
     switch (dir) {
-        case MovementDirection::FORWARD:
-            mEyePosition += float3(0, 0, distance);
+        case MovementDirection::FORWARD: {
+            float3 unitVector = glm::normalize(SphericalToCartesian(mTheta, mPhi));
+            mEyePosition += distance * unitVector;
             break;
-        case MovementDirection::LEFT:
-            mEyePosition += float3(distance, 0, 0);
+        }
+        case MovementDirection::LEFT: {
+            float3 perpendicularUnitVector = glm::normalize(SphericalToCartesian(mTheta + pi<float>() / 2.0f, pi<float>() / 2.0f));
+            mEyePosition -= distance * perpendicularUnitVector;
             break;
-        case MovementDirection::RIGHT:
-            mEyePosition += float3(-distance, 0, 0);
+        }
+        case MovementDirection::RIGHT: {
+            float3 perpendicularUnitVector = glm::normalize(SphericalToCartesian(mTheta + pi<float>() / 2.0f, pi<float>() / 2.0f));
+            mEyePosition += distance * perpendicularUnitVector;
             break;
-        case MovementDirection::BACKWARD:
-            mEyePosition += float3(0, 0, -distance);
+        }
+        case MovementDirection::BACKWARD: {
+            float3 unitVector = glm::normalize(SphericalToCartesian(mTheta, mPhi));
+            mEyePosition -= distance * unitVector;
             break;
+        }
+    }
+    mTarget = mEyePosition + SphericalToCartesian(mTheta, mPhi);
+    LookAt(mEyePosition, mTarget);
+}
+
+void FreeCamera::Turn(float deltaTheta, float deltaPhi)
```

Comment:
You should add a comment saying if theta/phi is doing latitude or longitude (conventions seems to differ)